### PR TITLE
✨ feat(mixins): Add-BaseLocation-abstract-model

### PR DESCRIPTION
**Pull request by ARYAN-NIKNEZHAD**
Update the structure of models

### CODE REVIEW

File: iranian_cities/mixins/base_location.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+from django.db.models import Model, CharField, BigIntegerField
+from django.utils.translation import gettext_lazy as _
+
+
+class BaseLocation(Model):
+    name = CharField(verbose_name=_("Name"), max_length=255)
```
Comment (sepehr-akbarzadeh) [inline iranian_cities/mixins/base_location.py]: - [ ] Each field must have `help_text` and `db_comment` attributes.

- [ ] `help_text` must be userfriendly for end user and `db_comment` is for developer that knows terminologies.

- [ ] So, don't forget to make `help_text` translatable

File: iranian_cities/models.py [hunk 2]
```diff
@@ -110,19 +85,13 @@ class RuralDistrict(models.Model):
         related_name='ruraldistricts',
```
Comment (sepehr-akbarzadeh) [inline iranian_cities/models.py]: Fix this related name also

**Review by sepehr-akbarzadeh — CHANGES_REQUESTED** [review]

File: iranian_cities/mixins/base_location.py (hunk 1 above)
Comment (ARYAN-NIKNEZHAD) [reply iranian_cities/mixins/base_location.py]: @sepehr-akbarzadeh, thank you for your valuable feedback! I've implemented the changes based on your suggestions. Please review the updated code and let me know if there are any further adjustments needed.

File: iranian_cities/models.py (hunk 2 above)
Comment (ARYAN-NIKNEZHAD) [reply iranian_cities/models.py]: @sepehr-akbarzadeh, thank you for your valuable feedback! I've implemented the changes based on your suggestions. Please review the updated code and let me know if there are any further adjustments needed.

**Pull request merged**